libraries/adaptive-expressions/src/functionUtils.ts

```diff
@@ -529,6 +531,32 @@ export class FunctionUtils {
         };
     }
 
+        /**
+     * Generate an expression delegate that applies function after verifying all children.
+     * @param func Function to apply.
```

Comment:
We should have a lint rule that requires that types are explicitly provided in JSDoc comments.